spatial/curve/hilbert_test.go

```diff
@@ -0,0 +1,346 @@
+package curve
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"golang.org/x/exp/rand"
+)
+
+type Dim int
```

Comment:
I don't think this type is more helpful than a function with an `int` parameter, or preferably a slice of test cases to iterate over.

Reply:
Part of the point of `testCurve` is to ensure that the curve actually describes a consistent curve where any two neighboring points are actually neighbors. To test this, I scan the entire space/curve. IMO a recursive function is the simplest way to do this. I replaced `Dim` with a function.

Reply:
Thanks.